### PR TITLE
Hubspot emails

### PR DESCRIPTION
**Pull request by gizemocak**
Screenshot 2020-07-21 at 16 04 37

Trello card: https://trello.com/c/CGCYWg3F/3719-4-add-a-cta-to-the-clausehound-blog

- Adds a Hubspot form for users to 'subscribe' to marketing emails. (Adds user emails to `Hubspot Contacts`)
- Adds the Hubspot chatbot to the blog.

### CODE REVIEW

File: src/components/hubspotForm.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          name: "email",
+          value: email,
+        },
+        {
+          name: "firstname",
+          value: name,
+        },
+      ],
+      context: {
+        pageUri: "https://blog.clausehound.com/",
```
Comment (theProf) [inline src/components/hubspotForm.ts:50]: do we want to show which specific page they are submitting from? or is this enough information?

File: src/components/hubspotForm.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    var final_data = JSON.stringify(data);
+
+    xhr.open("POST", url);
+    xhr.setRequestHeader("Content-Type", "application/json");
+    xhr.onreadystatechange = function() {
+      if (xhr.readyState === 4 && xhr.status === 200) {
+        setSubmitted(true);
+      }
+    };
+    xhr.send(final_data);
```
Comment (theProf) [inline src/components/hubspotForm.ts:64]: no need for 2 intermediate variables if this is the only place it gets used

File: src/components/hubspotForm.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        pageName: "Clausehound Blog",
+      },
+    };
+
+    var final_data = JSON.stringify(data);
+
+    xhr.open("POST", url);
+    xhr.setRequestHeader("Content-Type", "application/json");
+    xhr.onreadystatechange = function() {
+      if (xhr.readyState === 4 && xhr.status === 200) {
```
Comment (theProf) [inline src/components/hubspotForm.ts:60]: should also be some sort of user feedback on failures

File: src/styles/global.css [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#hubspot-messages-iframe-container > iframe {
```
Comment (theProf) [inline src/styles/global.css:1]: this should probably have some sort of comment above it as to why it's been added

File: src/templates/blog-post.ts [hunk 5]
```diff
@@ -86,6 +87,10 @@ const BlogPostTemplate: FC<Props> = ({ data, pageContext, location }) => {
       Helmet,
       null,
       h("script", { src: "https://www.clausehound.com/dist/clausehound.js" }),
+      h("script", {
+        id: "hs-script-loader",
+        src: `//js.hs-scripts.com/${process.env.HUBSPOT_ACCOUNT_NUMBER}.js`,
```
Comment (theProf) [inline src/templates/blog-post.ts:92]: is this for the form?

Comment (gizemocak) [reply src/templates/blog-post.ts:92]: this is for the hubspot chatbot

File: src/components/hubspotForm.ts (hunk 1 above)
Comment (gizemocak) [reply src/components/hubspotForm.ts:50]: actually a good idea to include the specific pages

**Pull request merged**